# Who's the Best NCAA Tournament Coach?

With the NCAA Men's basketball tournament and my bracket officially busted I want to start gathering information for next year. I unfortunately chose Virginia to win it all over Villanova as I thought they were the best teams all year long. I was right about one team but woefully wrong about the other. Congrats to UMBC but I want to win my bracket challenge next year.

One aspect I have thought to include in my choices next year is the coach of the team. It seems to me that the same coaches are in the sweet sixteen and beyond each year (Bill Self, Coach K, Jim Boeheim, etc). Also it seems certain coaches get their teams to overacheive (Beilein, Brad Stevens, etc.). My goal is to find current coaches that win more than they are supposed to in the tournament.

To do this I have taken the data from the [Kaggle NCAA 2018 ML Contest](https://www.kaggle.com/c/mens-machine-learning-competition-2018) and will analyze it to find the average amount of wins for each seed and compare that to how the coaches do.

## Import, Combine, and Clean the Data

To start I import the data for the conferences, teams, and coaches. Then I merge (AKA SQL join) and sort them by team to get data for each team, their coach, and the current conference on a yearly basis. 

In [53]:
%matplotlib inline
import matplotlib as plt
import pandas as pd
import numpy as np
import seaborn as sns
import scipy as spy
from sklearn.model_selection import train_test_split

conf = pd.read_csv("TeamConferences.csv")
teams = pd.read_csv("Teams.csv")
coaches = pd.read_csv("TeamCoaches.csv")

allTeams = coaches.merge(teams[["TeamID", "TeamName"]], on="TeamID").merge(conf, on=["Season", "TeamID"])
allTeams.drop(["FirstDayNum","LastDayNum"], axis=1, inplace=True)
allTeams.drop_duplicates(inplace=True)
allTeams.head(25)


Season  TeamID      CoachName   TeamName ConfAbbrev
0     1985    1102  reggie_minton  Air Force        wac
1     1986    1102  reggie_minton  Air Force        wac
2     1987    1102  reggie_minton  Air Force        wac
3     1988    1102  reggie_minton  Air Force        wac
4     1989    1102  reggie_minton  Air Force        wac
5     1990    1102  reggie_minton  Air Force        wac
6     1991    1102  reggie_minton  Air Force        wac
7     1992    1102  reggie_minton  Air Force        wac
8     1993    1102  reggie_minton  Air Force        wac
9     1994    1102  reggie_minton  Air Force        wac
10    1995    1102  reggie_minton  Air Force        wac
11    1996    1102  reggie_minton  Air Force        wac
12    1997    1102  reggie_minton  Air Force        wac
13    1998    1102  reggie_minton  Air Force        wac
14    1999    1102  reggie_minton  Air Force        wac
15    2000    1102  reggie_minton  Air Force        mwc
16    2001    1102      joe_scott  Air Force        mwc
17    2002    1102      joe_scott  Air Force        mwc
18    2003    1102      joe_scott  Air Force        mwc
19    2004    1102      joe_scott  Air Force        mwc
20    2005    1102   chris_mooney  Air Force        mwc
21    2006    1102   jeff_bzdelik  Air Force        mwc
22    2007    1102   jeff_bzdelik  Air Force        mwc
23    2008    1102  jeff_reynolds  Air Force        mwc
24    2009    1102  jeff_reynolds  Air Force        mwc

Next I will import the compact results as well as the tournament seeds files. Then merge the results with the seeds to create a data frame with all of the results as well as the winning seed and losing seed. I also had to clean the seed values as the seeds included a letter denoting which region the seed was in. All I cared about was the number of the seed, for more explanation and description of the function used see the [Basic Starter Kernel - NCAA Men's Dataset](https://www.kaggle.com/juliaelliott/basic-starter-kernel-ncaa-men-s-dataset).

In [54]:
results = pd.read_csv("NCAATourneyCompactResults.csv")
seeds = pd.read_csv("NCAATourneySeeds.csv")
import re

# convert and remove string characters from seed values
# borrowed from the Kaggle NCAA 2018 Men's competition tutorial notebook.
def seed_to_int(seed):
    #Get just the digits from the seeding. Return as int
    s_int = int(re.sub(r'[A-z]', '', seed))
    return s_int


seeds['seed_int'] = seeds.Seed.apply(seed_to_int)
seeds.drop(labels=['Seed'], inplace=True, axis=1)

seedResults =  results.merge(seeds, right_on=["TeamID","Season"], left_on= ["WTeamID","Season"]).merge(seeds, right_on=["TeamID","Season"], left_on= ["LTeamID","Season"])
seedResults.drop(["TeamID_x", "TeamID_y"],axis = 1, inplace=True)
seedResults.rename(index=str, columns={"seed_int_x":"WSeed", "seed_int_y":"LSeed"}, inplace=True)
seedResults.tail(10)



Season  DayNum  WTeamID  WScore  LTeamID  LScore WLoc  NumOT  WSeed  \
2170    2018     137     1345      76     1139      73    N      0      2   
2171    2018     135     1393      57     1395      52    N      0     11   
2172    2018     137     1393      55     1277      53    N      0     11   
2173    2018     135     1401      73     1344      69    N      0      7   
2174    2018     137     1401      86     1314      65    N      0      7   
2175    2018     135     1420      67     1438      54    N      0     16   
2176    2018     135     1452      85     1293      68    N      0      5   
2177    2018     137     1452      94     1267      71    N      0      5   
2178    2018     135     1267      81     1455      75    N      0     13   
2179    2018     135     1462     102     1411      83    N      0      1   

      LSeed  
2170     10  
2171      6  
2172      3  
2173     10  
2174      2  
2175      1  
2176     12  
2177     13  
2178      4  
2179     16

Next I needed to merge the seed results with the team information

In [55]:
allResults =  seedResults.merge(allTeams, right_on=["TeamID","Season"], left_on= ["WTeamID","Season"]).merge(allTeams, right_on=["TeamID","Season"], left_on= ["LTeamID","Season"])

def xyToWL(colNames):
    """
    Convert suffix _x and _y to prefix W and L respectively.

    Parameters
    ----------
    colNames : list
        List of column names to convert

    Returns
    -------
    list
        New column names converted to WColumnName from ColumnName_x

    """
    newCols = []
    for col in colNames:
        if col.endswith("_x"):
            newCol = col.replace("_x", "")
            newCol = "W" + newCol
            newCols.append(newCol)
        elif col.endswith("_y"):
            newCol = col.replace("_y", "")
            newCol = "L" + newCol
            newCols.append(newCol)
        else:
            newCols.append(col)
    return newCols

# drop repeated w team id and l team id
allResults.drop(["TeamID_x", "TeamID_y"], axis=1, inplace=True)
allResults.drop_duplicates(inplace=True)

# Convert suffixed columns
allResults.columns = xyToWL(allResults.columns)
allResults = allResults[~((allResults.WSeed ==16) & (allResults.LSeed == 16))]
allResults.sort_values(["Season","DayNum"], inplace=True)
allResults.tail(15)

Season  DayNum  WTeamID  WScore  LTeamID  LScore WLoc  NumOT  WSeed  \
2156    2018     141     1260      69     1305      68    N      0     11   
2160    2018     141     1276      99     1401      72    N      0      3   
2184    2018     141     1243      61     1246      58    N      0      9   
2188    2018     141     1199      75     1211      60    N      0      9   
2143    2018     142     1181      69     1393      65    N      0      2   
2150    2018     142     1242      80     1155      76    N      0      1   
2169    2018     142     1403      78     1345      65    N      0      3   
2172    2018     142     1437      90     1452      78    N      0      1   
2157    2018     143     1260      78     1243      62    N      0     11   
2161    2018     143     1276      58     1199      54    N      0      3   
2151    2018     144     1242      85     1181      81    N      1      1   
2173    2018     144     1437      71     1403      59    N      0      1   
2162    2018     150     1276      69     1260      57    N      0      3   
2174    2018     150     1437      95     1242      79    N      0      1   
2175    2018     152     1437      79     1276      62    N      0      1   

      LSeed        WCoachName       WTeamName WConfAbbrev        LCoachName  \
2156      7      porter_moser  Loyola-Chicago         mvc    eric_musselman   
2160      7      john_beilein        Michigan     big_ten     billy_kennedy   
2184      5       bruce_weber       Kansas St  big_twelve     john_calipari   
2188      4  leonard_hamilton      Florida St         acc          mark_few   
2143     11   mike_krzyzewski            Duke         acc       jim_boeheim   
2150      5         bill_self          Kansas  big_twelve     brad_brownell   
2169      2       chris_beard      Texas Tech  big_twelve      matt_painter   
2172      5        jay_wright       Villanova    big_east       bob_huggins   
2157      9      porter_moser  Loyola-Chicago         mvc       bruce_weber   
2161      9      john_beilein        Michigan     big_ten  leonard_hamilton   
2151      2         bill_self          Kansas  big_twelve   mike_krzyzewski   
2173      3        jay_wright       Villanova    big_east       chris_beard   
2162     11      john_beilein        Michigan     big_ten      porter_moser   
2174      1        jay_wright       Villanova    big_east         bill_self   
2175      3        jay_wright       Villanova    big_east      john_beilein   

           LTeamName LConfAbbrev  
2156          Nevada         mwc  
2160       Texas A&M         sec  
2184        Kentucky         sec  
2188         Gonzaga         wcc  
2143        Syracuse         acc  
2150         Clemson         acc  
2169          Purdue     big_ten  
2172   West Virginia  big_twelve  
2157       Kansas St  big_twelve  
2161      Florida St         acc  
2151            Duke         acc  
2173      Texas Tech  big_twelve  
2162  Loyola-Chicago         mvc  
2174          Kansas  big_twelve  
2175        Michigan     big_ten

Next I needed to calculate the expected wins (EWins) for each seed. To do this I grouped and merged the `allResults` dataframe on the seeds to give the `totalWins`, `totalLosses`, and `WPct` for each seed. I then calculated the `EWins` using the following formula:

$$EWins = \frac{totalWins}{4 \cdot numberOfSeasons}$$ 

*Note: The 4 in the denominator is due to the fact that each seed has 4 teams per year.*

In [56]:
# helper function mergeWL
def mergeWL(resDF, onCol):
    '''
    Group resDF by onCol and count totalWins, total Losses and WPct
    
    Parameters
    ----------
    resDF : pandas.dataframe
        Dataframe of results to filter and group wins and losses
        
    onCol: str
        String of column to group by

    Returns
    -------
    pandas.dataframe
        New dataframe with wins, losses, and WPct calculated.

    '''
    wins = resDF.groupby("W"+onCol, as_index=False)["WTeamID"].agg(['count']).rename(columns={'count': 'totalWins'})
    loss = resDF.groupby("L"+onCol, as_index=False)["LTeamID"].agg(['count']).rename(columns={'count': 'totalLosses'})
    comb = wins.merge(loss, how="right", left_index = True, right_index = True)
    comb.reset_index(inplace=True)
    comb.rename(columns={comb.columns[0]:onCol}, inplace=True)
    comb["WPct"] = comb.totalWins/(comb.totalWins + comb.totalLosses)
    comb.fillna(0, inplace=True)
    return comb

In [57]:
bySeed = mergeWL(allResults, "Seed")
# EWins is yearly expected wins per team by their seed. --> totalWins / 4 * number of seasons   (4 due to 4 different seeds every year)
bySeed["EWins"] = bySeed.totalWins/(4*seedResults.Season.nunique())
bySeed = bySeed.round(decimals=3)
bySeed.head(16)


Seed  totalWins  totalLosses   WPct  EWins
0      1        457          118  0.795  3.360
1      2        327          134  0.709  2.404
2      3        255          135  0.654  1.875
3      4        210          135  0.609  1.544
4      5        152          137  0.526  1.118
5      6        151          137  0.524  1.110
6      7        128          136  0.485  0.941
7      8         98          136  0.419  0.721
8      9         81          136  0.373  0.596
9     10         84          138  0.378  0.618
10    11         95          148  0.391  0.699
11    12         71          139  0.338  0.522
12    13         35          137  0.203  0.257
13    14         24          138  0.148  0.176
14    15          9          137  0.062  0.066
15    16          1          137  0.007  0.007

Finally on to calculating coach wins, losses, and the average of their wins added based on their seed (`AvgWinsAdded`). To start I grouped and merged `allResults` by `CoachName`.

In [58]:
byCoach = mergeWL(allResults, "CoachName")
byCoach.head(10)

CoachName  totalWins  totalLosses      WPct
0       al_brown        0.0            1  0.000000
1     al_skinner        8.0            9  0.470588
2   alan_leforce        1.0            2  0.333333
3   andrew_toole        0.0            1  0.000000
4   andy_enfield        4.0            3  0.571429
5   andy_kennedy        2.0            2  0.500000
6     andy_russo        2.0            2  0.500000
7   andy_stoglin        0.0            2  0.000000
8  anthony_evans        1.0            1  0.500000
9  anthony_grant        1.0            3  0.250000

Next I calculated the average seed (`avgSeed`) and number of appearances (`Appearances`) for each coach.

Finally I needed to calculate the number of wins added per year by each coach. To do this I created another dataframe called `byCoachSeed` to hold all of the wins and losses grouped by coach and season. Using this dataframe I was able to count the number of wins each year by each team (coach) to get their wins in the tournament for that year. Also I calculated the difference between average wins by seed and the actual wins that year for that coach(`Diff`).

In [59]:
winCoachSeed = allResults.groupby(["WCoachName", "Season"], as_index=False)["WSeed"].mean()
lossCoachSeed = allResults.groupby(["LCoachName", "Season"], as_index=False)["LSeed"].mean()
winCoachSeed.rename(columns={"WCoachName":"CoachName","WSeed":"Seed"}, inplace=True)
lossCoachSeed.rename(columns={"LCoachName":"CoachName","LSeed":"Seed"}, inplace=True)
byCoachSeed = winCoachSeed.append(lossCoachSeed,ignore_index=True).drop_duplicates().sort_values(["CoachName", "Season"])
byCoachSeed.reset_index(drop=True, inplace=True)

coachWins = allResults.groupby(["WCoachName", "Season"], as_index=False)["WSeed"].count().rename(columns={"WSeed":"TourneyWins","WCoachName":"CoachName"})
byCoachSeed = byCoachSeed.merge(coachWins, how="left",on=["CoachName", "Season"]).fillna(0)
byCoachSeed["TourneyWins"] = byCoachSeed["TourneyWins"].astype(int)

byCoachSeed = byCoachSeed.merge(bySeed[["Seed", "EWins"]], how="left", on="Seed").fillna(0)
byCoachSeed["Diff"] = byCoachSeed.TourneyWins - byCoachSeed.EWins
byCoachSeed.head(15)



CoachName  Season  Seed  TourneyWins  EWins   Diff
0       al_brown    1986    14            0  0.176 -0.176
1     al_skinner    1993     8            1  0.721  0.279
2     al_skinner    1997     9            0  0.596 -0.596
3     al_skinner    2001     3            1  1.875 -0.875
4     al_skinner    2002    11            0  0.699 -0.699
5     al_skinner    2004     6            2  1.110  0.890
6     al_skinner    2005     4            1  1.544 -0.544
7     al_skinner    2006     4            2  1.544  0.456
8     al_skinner    2007     7            1  0.941  0.059
9     al_skinner    2009     7            0  0.941 -0.941
10  alan_leforce    1991    10            0  0.618 -0.618
11  alan_leforce    1992    14            1  0.176  0.824
12  andrew_toole    2015    16            0  0.007 -0.007
13  andy_enfield    2013    15            2  0.066  1.934
14  andy_enfield    2016     8            0  0.721 -0.721

In [60]:
avgSeed = byCoachSeed.groupby("CoachName", as_index=False).agg({'Seed': "mean", 'Season': 'count'})
avgSeed.rename(columns={"Seed":"AvgSeed", "Season":"Appearances"}, inplace=True)
byCoach = byCoach.merge(avgSeed,on="CoachName").round(3)
byCoach["totalWins"] = byCoach["totalWins"].astype(int)
byCoach.head(10)

CoachName  totalWins  totalLosses   WPct  AvgSeed  Appearances
0       al_brown          0            1  0.000   14.000            1
1     al_skinner          8            9  0.471    6.556            9
2   alan_leforce          1            2  0.333   12.000            2
3   andrew_toole          0            1  0.000   16.000            1
4   andy_enfield          4            3  0.571   11.333            3
5   andy_kennedy          2            2  0.500   11.500            2
6     andy_russo          2            2  0.500    8.500            2
7   andy_stoglin          0            2  0.000   16.000            2
8  anthony_evans          1            1  0.500   15.000            1
9  anthony_grant          1            3  0.250   10.333            3

To get the `AvgWinsAdded` for each coach I was able to group `byCoachSeed` from above on the coaches 

In [61]:
coachTotalEWins = byCoachSeed.groupby("CoachName", as_index=False)["Diff"].mean()
byCoach = byCoach.merge(coachTotalEWins, how="left", on="CoachName").rename(columns={"Diff":"AvgWinsAdded"})
byCoach.sort_values("AvgWinsAdded",ascending=False, inplace=True)
byCoach.reset_index(drop=True, inplace=True)
byCoach.head(15)


CoachName  totalWins  totalLosses   WPct  AvgSeed  Appearances  \
0       porter_moser          4            1  0.800     11.0            1   
1      john_giannini          3            1  0.750     13.0            1   
2        kevin_ollie          7            1  0.875      8.0            2   
3     jim_rosborough          5            1  0.833      2.0            1   
4       kevin_mackey          2            1  0.667     14.0            1   
5       brad_stevens         12            5  0.706      7.0            5   
6   rollie_massimino         11            4  0.733      9.0            5   
7    jim_brandenburg          2            1  0.667     12.0            1   
8        darrin_horn          2            1  0.667     12.0            1   
9         joe_b_hall          2            1  0.667     12.0            1   
10      russ_pennell          2            1  0.667     12.0            1   
11    johnny_dawkins          2            1  0.667     10.0            1   
12    craig_esherick          2            1  0.667     10.0            1   
13    todd_lickliter          4            2  0.667      8.5            2   
14         greg_gard          4            2  0.667      7.5            2   

    AvgWinsAdded  
0         3.3010  
1         2.7430  
2         2.7315  
3         2.5960  
4         1.8240  
5         1.5028  
6         1.4866  
7         1.4780  
8         1.4780  
9         1.4780  
10        1.4780  
11        1.3820  
12        1.3820  
13        1.1800  
14        1.1690

Kevin Ollie leads the pack! He took a 7 seed to the Final Four and won it the next year. However I am more focused on coaches that have more lasting impact. Let's limit it to 3 tournament appearances.


### Top 50 Coaches by Average Wins Above  Seed 

In [62]:
byCoach3App = byCoach[byCoach["Appearances"] >= 3].reset_index(drop=True)
byCoach3App.head(50)

CoachName  totalWins  totalLosses   WPct  AvgSeed  Appearances  \
0          brad_stevens         12            5  0.706    7.000            5   
1      rollie_massimino         11            4  0.733    9.000            5   
2          john_beilein         24           12  0.667    7.417           12   
3           larry_brown         13            4  0.765    4.200            5   
4      richard_williams          6            3  0.667    5.000            3   
5            john_groce          4            3  0.571   11.333            3   
6          andy_enfield          4            3  0.571   11.333            3   
7          frank_martin         10            5  0.667    6.600            5   
8         billy_donovan         35           12  0.745    4.071           14   
9           sonny_smith          7            5  0.583    9.400            5   
10        paul_westhead          4            3  0.571   11.000            3   
11       bill_guthridge          8            3  0.727    4.000            3   
12           stan_heath          5            4  0.556   10.500            4   
13          rick_pitino         54           18  0.750    3.550           20   
14             tom_izzo         48           21  0.696    4.857           21   
15        john_calipari         54           18  0.750    3.000           19   
16          quin_snyder          5            4  0.556    9.000            4   
17      jerry_tarkanian         22            8  0.733    4.111            9   
18        p_j_carlesimo         12            7  0.632    5.000            6   
19       lefty_driesell          5            4  0.556    8.750            4   
20        billy_kennedy          5            4  0.556    9.500            4   
21           denny_crum         21           11  0.656    5.583           12   
22         dick_tarrant          3            4  0.429   13.250            4   
23          jim_calhoun         48           17  0.738    4.500           20   
24        archie_miller          5            4  0.556    9.000            4   
25           dean_smith         37           13  0.740    2.615           13   
26         steve_fisher         26           14  0.650    6.267           15   
27         tommy_amaker          4            5  0.444   12.200            5   
28           dan_monson          3            3  0.500   10.000            3   
29       thomas_penders         12           11  0.522    9.636           11   
30         roy_williams         78           25  0.757    2.714           28   
31        jim_j_o'brien         11            7  0.611    5.857            7   
32              bo_ryan         27           15  0.643    5.200           15   
33         clem_haskins         11            7  0.611    6.714            7   
34      mike_krzyzewski         95           29  0.766    2.212           33   
35          jim_boeheim         54           26  0.675    4.296           27   
36          pete_gillen          8           10  0.444   10.444            9   
37          paul_hewitt          7            6  0.538    8.167            6   
38  nolan_richardson_jr         26           13  0.667    5.500           14   
39        gary_williams         28           15  0.651    5.375           16   
40         bill_frieder         14            6  0.700    4.429            7   
41        steve_donahue          2            4  0.333   14.000            4   
42           john_brady          6            4  0.600    5.500            4   
43        fang_mitchell          1            3  0.250   15.000            3   
44         ben_jacobson          4            4  0.500    9.250            4   
45           dick_davey          2            3  0.400   12.333            3   
46          jim_valvano          8            5  0.615    5.600            5   
47          ben_howland         19           10  0.655    5.500           10   
48          gary_waters          2            3  0.400   12.333            3   
49       wimp_

### Bottom 50 Coaches 

In [63]:
byCoach3App.tail(50)

CoachName  totalWins  totalLosses   WPct  AvgSeed  Appearances  \
202        mick_cronin          6           10  0.375    7.800           10   
203          dave_odom         10            9  0.526    5.222            9   
204     seth_greenberg          1            3  0.250    9.667            3   
205         bobby_lutz          2            5  0.286    7.800            5   
206    mike_montgomery         18           16  0.529    6.188           16   
207           mark_fox          2            5  0.286    8.200            5   
208    kevin_stallings          6            9  0.400    6.778            9   
209       wayne_tinkle          0            4  0.000   11.750            4   
210         tom_asbury          0            4  0.000   12.250            4   
211        fran_dunphy          3           16  0.158   10.625           16   
212     kelvin_sampson         13           15  0.464    6.400           15   
213       ray_mccallum          0            3  0.000   12.667            3   
214          tad_boyle          1            4  0.200    9.250            4   
215      speedy_morris          1            4  0.200    9.500            4   
216      jud_heathcote          7            7  0.500    5.143            7   
217       kevin_keatts          0            3  0.000   11.333            3   
218        gene_bartow          2            5  0.286    8.200            5   
219      riley_wallace          0            3  0.000   11.667            3   
220      kevin_willard          1            3  0.250    7.667            3   
221     larry_eustachy          4            5  0.444    6.800            5   
222        randy_ayers          6            3  0.667    3.333            3   
223     greg_mcdermott          3            8  0.273    8.375            8   
224     rick_stansbury          4            6  0.400    6.667            6   
225         gene_keady         18           15  0.545    4.933           15   
226        rich_herrin          0            3  0.000   11.667            3   
227        travis_ford          1            6  0.143    8.833            6   
228    steve_cleveland          0            3  0.000   12.000            3   
229         bob_wenzel          0            3  0.000   10.000            3   
230        jerry_green          3            5  0.375    6.000            5   
231       skip_prosser          6            9  0.400    7.222            9   
232         lou_henson          9           11  0.450    5.800           10   
233          ed_cooley          1            5  0.167    9.400            5   
234        jamie_dixon         12           13  0.480    5.167           12   
235  john_thompson_iii          9           10  0.474    5.800           10   
236        billy_tubbs         14            8  0.636    2.875            8   
237        hugh_durham          1            4  0.200    8.000            4   
238         pat_foster          0            3  0.000   10.000            3   
239          rob_evans          1            3  0.250    7.333            3   
240       norm_stewart          7           10  0.412    5.900           10   
241       jeff_mullins          0            3  0.000    9.000            3   
242       ralph_miller          0            3  0.000    9.333            3   
243          tim_welsh          0            3  0.000    9.000            3   
244    george_raveling          1            4  0.200    7.750            4   
245          danny_nee          0            6  0.000    8.667            6   
246     oliver_purnell          0            6  0.000    8.167            6   
247        j_d_barnett          1            3  0.250    7.667            3   
248        frank_haith          1            4  0.200    7.250            4   
249       tony_bennett         10            8  0.556    3.375            8   
250       eddie_fogler          2            6  0.250    6.667            6   
251       steve_lappas          2            4  0.333    4.500 

## Conclusion 

Looking at the Top 50 and bottom 50 provide some insights. Tony Bennett is in the bottom 10% (234 out of 247). Maybe defense doesn't win championships? Also if Pitt wasn't happy with Jamie Dixon and his teams' tournament performance, Kevin Stallings might not have been much of an improvement. On the positive side I can see why Brad Stevens has had success in both NCAA and NBA as he leads the group of coaches with 3 or more appearances. Also a lot of the legends (Izzo, Pitino, Calipari) are up towards the top as well.

Some teams to watch next year if they make the tournament could be Indiana (Archie Miller), South Carolina (Frank Martin), and Michigan (John Beilein). It would be nice to add the results from this year's tournament as some of these values might change. Beilein would improve as he took #3 Michigan to the Final and Tony Bennett will fall even further.

**Update** I added the data for the 2018 tournament and John Beilein did indeed improve all the way up to the number three spot. Unfortunately Tony Bennett dropped significantly due to losing in the first round as a number one seed.